### ISSUE TITLE
NVDA fails to read checked or unchecked in Server Manager while installing roles or features

### ISSUE THREAD

**Issue by Cambridgeport90** [nvaccess/nvda#7529]
To reproduce this:
Open Server Manager from a Windows server 2012 or newer server
click add roles and features
select a role or feature to be installed
Notice on the next screen that NVDA reads the names, but does not read whether they are unchecked (not yet installed) or checked (installed). 
As a result, we are forced to use our judgements to ensure that everything gets installed correctly. This also applies to subfeatures.

**Comment by josephsl** [comment]
Hi, dev info please. Note that for others trying to reproduce this, you need to be using Server with desktop experience or minimal GUI (or if using Server 2016, use it on a server configured with full desktop experience installation). Thanks.

**Comment by Cambridgeport90** [comment]
What additional info do you need on this side? I've noticed this with all NVDA versions, but the one I'm running on the server is 2017.2 (couldn't get .3 to update).

**Comment by leonardder** [comment]
What josephsl meant: please provide some developer info about the controls in this dialog. For example, when one of the inaccessible controls is the navigator object (which is the case by default when it is focused), please press nvda+f1 and copy the part of the log file that starts with "Developer info for navigator object" to this thread.

**Comment by josephsl** [comment]
Hi,

A clearer STR:

1. From Server Manager, go to Manage/Add.
2. From the wizard that appears, you need to click Next two or three times until you arrive at the desired role (say, Active Directory Domain Services, Windows Server Update Services and so on).
3. Select a role, then go through dependencies/components.

Expected: NVDA should announce the state of the role installation.
Actual: no such announcement. The toggle control is a child of the dependency item, so one needs to locate this toggle button via object navigation and activate it to toggle it properly.
Cause: UIA encoding problem, ultimately traceable to use of Windows Presentation Foundation (WPF). Unfortunately, due to the age of the OS, the only thing we can do is report this to MS in hopes of getting this fixed before end of this year, but it is highly unlikely this will be looked into by the server and Azure teams.

Setup:

* NVDA version: next.14342.
* OS: Windows Server 2012 R2

Developer info:

name: u''
role: ROLE_TOGGLEBUTTON
states: 
isFocusable: False
hasFocus: False
Python object: <NVDAObjects.UIA.UIA object at 0x045A7210>
Python class mro: (<class 'NVDAObjects.UIA.UIA'>, <class 'NVDAObjects.window.Window'>, <class 'NVDAObjects.NVDAObject'>, <class 'baseObject.ScriptableObject'>, <class 'baseObject.AutoPropertyObject'>, <type 'object'>)
description: ''
location: (365, 366, 18, 21)
value: None
appModule: <'appModuleHandler' (appName u'servermanager', process ID 2392) at address 45c50f0>
appModule.productName: u'Microsoft\xae Windows\xae Operating System'
appModule.productVersion: u'6.3.9600.17238'
TextInfo: <class 'NVDAObjects.NVDAObjectTextInfo'>
windowHandle: 721426
windowClassName: u'HwndWrapper[ServerManager.exe;;ca54404a-de83-4cf6-8582-78f4cadda943]'
windowControlID: 0
windowStyle: 382205952
windowThreadID: 2404
windowText: u'Add Roles and Features Wizard'
displayText: u''
UIAElement: <POINTER(IUIAutomationElement) ptr=0x5e04ab0 at 45966c0>
UIA automationID: Expander
UIA frameworkID: WPF
UIA runtimeID: (7, 2392, 43001397)
UIA providerDescription: [pid:2392,hwnd:0x0 Main(parent link):Unidentified Provider (managed:MS.Internal.Automation.ElementProxy, PresentationCore, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)]
UIA className: Button

Our experience with WPF is a mixed one. Sometimes, it works, sometimes it poses performance issues, sometimes there's no workaround, and sometimes, UIA is broken in some apps powered by it. Unfortunately, it is highly unlikely that this problem could be fixed on time, otherwise we need to come up with a hack (again).

Thanks.

**Comment by leonardder** [comment]
I assume it is very unlikely that using IAccessible for these controls will improve things?

**Comment by josephsl** [comment]
Hi, I think @MichaelDCurran would know better. Thanks.

**Comment by Adriani90** [comment]
@Cambridgeport90 could you please test with NVDA 2019.1 if this issue is still occuring? if so, could you please upload the log file which you generate by pressing nvda+f1? Thanks!

**Comment by Cambridgeport90** [comment]
I can certainly do that for you. I actually had forgotten I’d entered this issue, so thanks for the reminder!

**Comment by josephsl** [comment]
Hi Kat, if you don’t mind, can you try duplicating this in Server 2019 as well (with Desktop Experience installed)? Thanks.

**Comment by Cambridgeport90** [comment]
Will do, certainly! I don’t currently have a Windows server 2019 install up to play with at the moment, but I will ideally shortly.

**Comment by Adriani90** [comment]
@Cambridgeport90 a further reminder for this issue, could you please thest with NVDA 2020.1 RC1 if this issue is still occuring? Thanks.

**Comment by Cambridgeport90** [comment]
I can do that.

**Comment by Adriani90** [comment]
@Cambridgeport90 please report your results with NVDA 2022.4. Is this issue still occuring for you?

**Comment by Adriani90** [comment]
We don't have any updates from the initial author after requesting info multiple times. Closing as abandoned. if you are still having this issue, please comment and we can reopen.

**Comment by Cambridgeport90** [comment]
I forgot that even existed in this situation. Thanks for the reminder!

**Comment by josephsl** [comment]
Hi,

June 2024 update: anyone want to test this on Server 2025 (preview)? Ping @Cambridgeport90 in case you are open to remote troubleshooting session with me or someone (note that I'm using Server 2025 Insider canary builds for testing purposes).

Thanks.

**Comment by Adriani90** [comment]
Is this still an issue with NvDA 2025.1.2 and Server manager 2025?

**Comment by Adriani90** [comment]
No updates upon request since more than 1 year, closing as abandoned. Please comment in case this issue still occurs and we can reopen.
